modules/ducktests/tests/ignitetest/tests/dns_failure_test.py

```diff
@@ -0,0 +1,134 @@
+    """
+
+    @cluster(num_nodes=4)
+    @ignite_versions(str(DEV_BRANCH))
+    def dns_failure_test(self, ignite_version):
+        """
+        DNS failure test.
+        """
+        # Replace hosts with IP addresses.
+        for node in self.test_context.cluster.nodes:
+            node.account.externally_routable_ip = socket.gethostbyname(node.account.externally_routable_ip)
+
+        version = IgniteVersion(ignite_version)
+
+        ignite_config = IgniteConfiguration(
+            version=version,
+            data_storage=DataStorageConfiguration(
+                default=DataRegionConfiguration(
+                    persistence_enabled=True
+                )
+            )
+        )
+
+        ignites = self.__prepare_service(ignite_config, 3)
+
+        self.__unblock_dns(ignites)
+
+        # Start nodes one-by-one to reproduce the problem.
+        ignites.start_node(ignites.nodes[0])
+        ignites.await_started([ignites.nodes[0]])
+        ignites.start_node(ignites.nodes[1])
+        ignites.await_started([ignites.nodes[0], ignites.nodes[1]])
+        ignites.start_node(ignites.nodes[2])
+        ignites.await_started()
+
+        control_utility = ControlUtility(ignites)
+        control_utility.activate()
+
+        self.__block_dns(ignites, 20000, True)
+
+        ignites.stop_node(ignites.nodes[1])
+
+        ignites.await_event("Node left topology", 60, from_the_beginning=True,
+                            nodes=[ignites.nodes[0], ignites.nodes[2]])
+
+        assert ignites.alive(ignites.nodes[0]), 'Node 0 should be alive'
+        assert ignites.alive(ignites.nodes[2]), 'Node 2 should be alive'
+
+        self.__unblock_dns(ignites, [ignites.nodes[1]])
+
+        ignites.start_node(ignites.nodes[1])
+        ignites.await_started()
+
+        assert len(ignites.alive_nodes) == 3, 'All nodes should be alive'
+
+        # Smoke test on full topology.
+        app = IgniteApplicationService(
+            self.test_context,
+            ignite_config._replace(client_mode=True, discovery_spi=from_ignite_cluster(ignites)),
+            startup_timeout_sec=120,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.smoke_test.SimpleApplication")
+
+        app.start()
+        app.stop()
+
+        ignites.stop()
+
+    def __prepare_service(self, ignite_config, num_nodes=1):
+        ignite = IgniteService(
+            self.test_context,
+            ignite_config,
+            startup_timeout_sec=120,
+            num_nodes=num_nodes)
+
+        bootclasspath = list(map(lambda lib: os.path.join(lib, "classes"), ignite.spec._module_libs("ducktests")))
+
+        # Note: Support of impl.prefix property was removed since java 18.
+        ignite.spec.jvm_opts.append("-Dimpl.prefix=BlockingDns")
+        ignite.spec.jvm_opts.append("-Xbootclasspath/a:" + ":".join(bootclasspath))
+
+        return ignite
+
+    @staticmethod
+    def __block_dns(ignite, timeout, fail, nodes=None):
```

Comment:
This is not a regression test. This test is to show that problem exists and problem is fixed by the patch. I've added this option to test, but don't sure it's necessary, since we can always run this test manually with fail=false option. Result is the same.